bin/functions.sh

```diff
@@ -20,6 +26,46 @@ function stat_size () {
   stat $stat_format_arg $stat_size_spec "$1" 2>/dev/null
 }
 
+function sha512sum() {
+  # Can't figure out a better way of escaping $_sha512sum for use in a call than using sh -c
+  sh -c "$_sha512sum $1|cut -d' ' -f1"
+}
+
+function escape_path() {
+  local r=$1
+  local host="$(host_os)"
+  if [[ "$host" == "Win32" ]]; then
+    r=${r//\\/\\\\}
+  fi
+  echo "$r"
+}
+
+function unix_path() {
+  local p="$(escape_path "$1")"
+  local host="$(host_os)"
+  if [[ "$host" == "Win32" ]]; then
+    p="$(cygpath -u "$p")"
```

Comment:
Is this portable?

Reply:
Currently we require mingw. I wouldn't call it portable.
Can we backlog this if you want to implement it in pure bash?